### DISCUSSION TITLE
Quantum Phase Estimation and Hadamard test

### DISCUSSION THREAD

**Discussion by vtomole** [vtomole/vtomole.github.io#49] [General]
Post: https://vtomole.github.io/blog/2018/05/20/pea

**Comment by masaduzz1** [comment]
Hi, thanks for the nice post. In the example of the Hadamard test, the output expression of the CU applied on 2 qubits does not look correct.

**Comment by vtomole** [comment]
Which part are you talking about exactly? Is it 

The Hadamard test solves: Given a Unitary  ....

part? What's not correct about it?

**Comment by masaduzz1** [comment]
Hi, thanks for the reply. When CU applied on (|0>+|1>) x |1>, isn't the o/p be |0>|1> + |1>(U|1>) = |0>|1> + |1>|0>?  I didn't carry the 1/sqrt(2) factor here.

**Comment by vtomole** [comment]
> |0>|1> + |1>(U|1>) = |0>|1> + |1>|0>


You have `U` on the left side but not the right. Where did it go?

**Comment by masaduzz1** [comment]
I thought only the second qubit is modified if 1st qubit is 1 when two qubits are applied in the CU gate. Please see wiki here: https://en.wikipedia.org/wiki/Quantum_logic_gate#Controlled_(cX_cY_cZ)_gates


Sincerely,

Asaduzzaman

________________________________

**Comment by vtomole** [comment]
You wrote

> |0>|1> + |1>(U|1>) = |0>|1> + |1>|0>

What is `U` here? Is it an arbitrary unitary or did you make it the X gate?

**Comment by masaduzz1** [comment]
I was talking about the hadamard test example in your post, U there is (0, 1; 1, 0)

________________________________

**Comment by vtomole** [comment]
Ah okay. I see my mistake. Thanks for pointing this out. I'll post a fix soon.

**Comment by masaduzz1** [comment]
great then, thank you.

________________________________

**Comment by vtomole** [comment]
@masaduzz1 Hey, i made the fixes. Can you check to make sure that it makes sense now? Also, i would like to send you $5 as a thank you for finding this. How should i go about this?
